File: API_SC_01_08Abril/API-Karoline-Raiane-da-Silva-Nascimento/q3_media.py
def main():

    sexo = input("Masculino ou Feminino?? (M ou F) ")
    nota = float(input("Nota: "))

    total_alunos = 0
    total_notas = 0
    maior_nota = nota
    menor_nota = nota
    total_masculino = 0
    total_feminino = 0
    homens = 0
    mulheres = 0


    while sexo == "F" or sexo == "M":
        if nota > maior_nota:
            maior_nota = nota
        if nota < menor_nota:
            menor_nota = nota

        if sexo == "F":
            total_feminino += nota
            mulheres += 1

        if sexo == "M":
            total_masculino += nota
            homens += 1

        total_notas += nota
        total_alunos += 1

        sexo = input("Masculino ou Feminino?? (M ou F) ")

        if sexo == "F" or sexo == "M":
            nota = float(input("Nota: "))
        else: 
            print("Fim")


    

    media = total_notas / total_alunos
    perf_masc = (total_masculino / total_notas) * 100
    perf_feminina = (total_feminino / total_notas) * 100
    desempenho_masc = total_masculino / homens
    desempenho_fem = total_feminino / mulheres
    class_geral, class_fem, class_masc = calc_desempenho(media, desempenho_masc, desempenho_fem)

    resultado = f""" 
    - Total de alunos: {total_alunos}
    - Total mulheres: {mulheres}
    - Total homens: {homens}
    - Maior nota geral: {maior_nota}
    - Menor nota geral: {menor_nota}
    - Média geral: {media} 
    - Classificação geral: {class_geral}
    - Performance feminina: {perf_feminina:.2f} %
    - Classificação feminina: {class_fem}
    - Performance masculina: {perf_masc:.2f} %
    - Classificação masculina: {class_masc}"""

    print(resultado)


def calc_desempenho(media, desempenho_masc, desempenho_fem):
    class_geral = ""
    class_masc = ""
    class_fem = ""

    if media <= 2:
        class_geral = "Péssimo"
    elif media <= 4:
        class_geral = "Ruim"
    elif media <= 7:
        class_geral = "Regular"
    elif media <= 8:
        desempenho_geral = "Bom"
    elif media <= 10:
        class_geral = "Excelente"

    if desempenho_fem <= 2:
        class_fem = "Péssimo"
    elif desempenho_fem <= 4:
        class_fem = "Ruim"
    elif desempenho_fem <= 7:
        class_fem = "Regular"
    elif desempenho_fem <= 8:
        class_fem = "Bom"
    elif desempenho_fem <= 10:
        class_fem = "Excelente"

    if desempenho_masc <= 2:
        class_masc = "Péssimo"
    elif desempenho_masc <= 4:
        class_masc = "Ruim"
    elif desempenho_masc <= 7:
        class_masc = "Regular"
    elif desempenho_masc <= 8:
        class_masc = "Bom"
    elif desempenho_masc <= 10:
        class_masc = "Excelente"

    return class_geral, class_fem, class_masc



    

main()

File: API_SC_01_08Abril/Isaac Borges/q3_media.js
import { get_entry, get_number } from "./api_utils.js"

//FUNÇÃO DE TRABALHO
function get_media(notatotal, n_alunos){
    return notatotal / n_alunos

}

function get_desempenho(media){
    let desempenho = 'Sem informação'

    if (media < 2){
        desempenho = 'PÉSSIMO'
    } else if (media < 4) {
        desempenho = 'RUIM'
    } else if (media < 7) {
        desempenho = 'REGULAR'
    } else if (media < 8) {
        desempenho = 'BOM'
    } else if (media <= 10) {
        desempenho = 'EXCELENTE'
    }

    return desempenho
}


// FUNÇÃO DE ENTRADA ADICIONADA NESSA QUESTÃO
function get_number_in_range(mensagem, min, max){
    const numero = get_number(mensagem)
    
    if (numero < min || numero > max){
        console.log('Número inválido!')
        return get_number_in_range(mensagem, min, max)
    }

    return numero
}   

function enter_to_continue(entry){
    const enter = get_entry(entry)

    return enter
}


// MAIN
function main(){
    console.log('   COMPUTAR NOTAS DOS ALUNOS NA PROVA   ')

    //Notas variaveis
    let nota1, media_geral, performance_f, performance_m, media_m, media_f
    //maior e menor
    let maior_nota = 0
    let menor_nota = 10
    //totais
    let notatotal = 0
    let notatotal_m = 0
    let notatotal_f = 0
    // Contagens 
    let num_alunos = 0
    let num_mulheres = 0 
    let num_homens = 0
    // Variaveis de mensagem 
    let msg_sexo = ''
    let msg_parcial 

    let sexo = 'M'
    sexo = get_entry('Qual o sexo do(a) aluno(a)? ("M" = masculino; "F" = feminino)   : ')
    while (sexo === 'M' || sexo === 'F'){
        
        if (sexo === 'M'){ // Mudar a proxima mensagem de acordo com o sexo indicado
            msg_sexo = 'do aluno'
        } else {
            msg_sexo = 'da aluna'
        }
        //PEDIR E CALCULAR A NOTATOTAL
        nota1 = get_number_in_range(`Qual a nota ${msg_sexo}? : `, 0, 10)
        notatotal += nota1
        num_alunos++
        
        if (sexo === 'F'){
            num_mulheres++
            notatotal_f += nota1
            
        } else {
            num_homens++
            notatotal_m += nota1
        }

        //CALCULAR MÉDIA
        media_geral = get_media(notatotal, num_alunos)
        media_m = get_media(notatotal_m, num_homens)
        performance_m = media_m * 10
        media_f = get_media(notatotal_f, num_mulheres)
        performance_f = media_f * 10

        // Ver qual a maior nota 
        if (nota1 >= maior_nota){
            maior_nota = nota1
        } if (nota1 <= menor_nota){
            menor_nota = nota1
        }

        msg_parcial =  `
        --------------------------------------
    
            Maior nota geral      : ${maior_nota.toFixed(1)}
            Menor nota geral      : ${menor_nota.toFixed(1)}
            Média geral           : ${media_geral.toFixed(1)}
            Performance masculina : ${performance_m.toFixed(1)}%
            Performance feminina  : ${performance_f.toFixed(1)}%
        
            PRESSIONE ENTER 
    
        --------------------------------------
        `
        //
        console.log(msg_parcial)
        enter_to_continue('')
        console.clear()

        sexo = get_entry('Qual o sexo do(a) aluno(a)? ("M" = masculino; "F" = feminino)   : ')
    }

    console.clear()

    const desempenho_turma = get_desempenho(media_geral)
    const desempenho_homens = get_desempenho(media_m)
    const desempenho_mulheres = get_desempenho(media_f)

    const msg_final = `
    --------------------------------------
        RELATÓRIO FINAL

        Maior nota geral      : ${maior_nota.toFixed(1)}
        Menor nota geral      : ${menor_nota.toFixed(1)}
        Média geral           : ${media_geral.toFixed(1)}
        Performance masculina : ${performance_m.toFixed(1)}%
        Performance feminina  : ${performance_f.toFixed(1)}%
    
        Número de alunos      : ${num_alunos}
        Número de homens      : ${num_homens}
        Número de mulheres    : ${num_mulheres}

        Desempenho da turma   : ${desempenho_turma}
        Desempenho masculino  : ${desempenho_homens}
        Desempenho feminino   : ${desempenho_mulheres}

    --------------------------------------
    `
    
    console.log(msg_final)
}



main()